### ISSUE TITLE
enhanced enums support for `avoid_renaming_method_parameters`

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#3095]

**Comment by pq** [comment]
```dart
  test_rename() async {
    await assertDiagnostics(r'''
class C {
  int f(int f) => f;
}
enum A extends C {
  a,b,c;
  @override
  int f(int x) => x;
}
''', [
      lint('avoid_renaming_method_parameters', 28, 8), // todo(pq): update index
    ]);
  }
```

Tip-of-trunk fails w/:

```
===== asynchronous gap ===========================
test/rules/avoid_renaming_method_parameters.dart 26:5  AvoidRenamingMethodParametersTest.test_rename

Expected but did not find:
  avoid_renaming_method_parameters [28, 8]

Found but did not expect:
  HintCode.UNUSED_ELEMENT [79, 1, The declaration 'f' isn't referenced.]
  ParserErrorCode.MISSING_IDENTIFIER [57, 1, Expected an identifier.]
  CompileTimeErrorCode.DUPLICATE_DEFINITION [48, 1, The name 'C' is already defined.]
  ParserErrorCode.MISSING_IDENTIFIER [55, 1, Expected an identifier.]
  ParserErrorCode.MISSING_ENUM_BODY [40, 7, An enum definition must have a body with at least one constant name.]
  ParserErrorCode.EXPECTED_TOKEN [40, 7, Expected to find ';'.]
  ParserErrorCode.MISSING_CONST_FINAL_VAR_OR_TYPE [40, 7, Variables must be declared using the keywords 'const', 'final', 'var' or a type name.]
  CompileTimeErrorCode.UNDEFINED_IDENTIFIER [58, 1, Undefined name 'c'.]
  ParserErrorCode.UNEXPECTED_TOKEN [55, 1, Unexpected text ';'.]
  ParserErrorCode.EXPECTED_TOKEN [54, 1, Expected to find ';'.]
  CompileTimeErrorCode.UNDEFINED_IDENTIFIER [56, 1, Undefined name 'b'.]
  ParserErrorCode.MISSING_FUNCTION_PARAMETERS [48, 1, Functions must have an explicit list of parameters.]
  ParserErrorCode.EXPECTED_IDENTIFIER_BUT_GOT_KEYWORD [40, 7, 'extends' can't be used as an identifier because it's a keyword.]
  CompileTimeErrorCode.UNDEFINED_IDENTIFIER [54, 1, Undefined name 'a'.]
  ParserErrorCode.EXPECTED_TOKEN [56, 1, Expected to find ';'.]
  ParserErrorCode.UNEXPECTED_TOKEN [57, 1, Unexpected text ';'.]

...
```

/fyi @scheglov
